src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -0,0 +1,220 @@
+    InstructionSet isa         = compiler->isaOfHWIntrinsic(intrinsicID);
+    switch (isa)
+    {
+        case InstructionSet_SSE:
+            genSSEIntrinsic(treeNode);
+            break;
+        case InstructionSet_SSE2:
+            genSSE2Intrinsic(treeNode);
+            break;
+        case InstructionSet_SSE3:
+            genSSE3Intrinsic(treeNode);
+            break;
+        case InstructionSet_SSSE3:
+            genSSSE3Intrinsic(treeNode);
+            break;
+        case InstructionSet_SSE41:
+            genSSE41Intrinsic(treeNode);
+            break;
+        case InstructionSet_SSE42:
+            genSSE42Intrinsic(treeNode);
+            break;
+        case InstructionSet_AVX:
+            genAVXIntrinsic(treeNode);
+            break;
+        case InstructionSet_AVX2:
+            genAVX2Intrinsic(treeNode);
+            break;
+        case InstructionSet_AES:
+            genAESIntrinsic(treeNode);
+            break;
+        case InstructionSet_BMI1:
+            genBMI1Intrinsic(treeNode);
+            break;
+        case InstructionSet_BMI2:
+            genBMI2Intrinsic(treeNode);
+            break;
+        case InstructionSet_FMA:
+            genFMAIntrinsic(treeNode);
+            break;
+        case InstructionSet_LZCNT:
+            genLZCNTIntrinsic(treeNode);
+            break;
+        case InstructionSet_PCLMULQDQ:
+            genPCLMULQDQIntrinsic(treeNode);
+            break;
+        case InstructionSet_POPCNT:
+            genPOPCNTIntrinsic(treeNode);
+            break;
+        default:
+            unreached();
+            break;
+    }
+}
+
+void CodeGen::genSSEIntrinsic(GenTreeHWIntrinsic* treeNode)
+{
+    // TODO - Implement SSE intrinsic code generation
+}
+
+void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* treeNode)
+{
+    // TODO - Implement SSE2 intrinsic code generation
+}
+
+void CodeGen::genSSE3Intrinsic(GenTreeHWIntrinsic* treeNode)
+{
+    // TODO - Implement SSE3 intrinsic code generation
+}
+
+void CodeGen::genSSSE3Intrinsic(GenTreeHWIntrinsic* treeNode)
+{
+    // TODO - Implement SSSE3 intrinsic code generation
+}
+
+void CodeGen::genSSE41Intrinsic(GenTreeHWIntrinsic* treeNode)
+{
+    // TODO - Implement SSE41 intrinsic code generation
+}
+
+void CodeGen::genSSE42Intrinsic(GenTreeHWIntrinsic* treeNode)
+{
+    NamedIntrinsic intrinsicID = treeNode->gtHWIntrinsicId;
+    GenTree*       op1         = treeNode->gtGetOp1();
+    GenTree*       op2         = treeNode->gtGetOp2();
+    regNumber      targetReg   = treeNode->gtRegNum;
+    assert(targetReg != REG_NA);
+    var_types targetType = treeNode->TypeGet();
+    var_types baseType   = treeNode->gtBaseType;
+
+    regNumber op1Reg = op1->gtRegNum;
+    regNumber op2Reg = op2->gtRegNum;
+
+    switch (intrinsicID)
+    {
+        case NI_SSE42_Crc32:
+            if (op1Reg != targetReg)
+            {
+                inst_RV_RV(ins_Copy(targetType), targetReg, op1Reg, targetType, emitTypeSize(targetType));
```

Comment:
You should just use `INS_mov`. Codegen has a habit of using `ins_Copy` and a few other similar functions in places where's no need and CPU cycles are wasted checking if the type is SIMD or floating point.

Reply:
Got it, will do.